src/lib/filehandler.rs

```diff
@@ -1,16 +1,52 @@
-use std::fs::File;
+use std::fs::*;
+use std::io::*;
+use basic_types::instruction_set::*;
+use basic_types::instruction::*;
+#[derive(Debug)]
 pub struct FileHandler {
     path: String,
-    file: File,
+    buf: BufReader<File>,
 }
 impl FileHandler {
     pub fn new(path: String) -> FileHandler {
         let file = File::open(&path).unwrap();
+        let f = BufReader::new(file);
         return FileHandler {
             path: path,
-            file: file,
+            buf: f,
         };
     }
+    pub fn read_instruction(&mut self) -> Option<Result<String>> {
```

Comment:
pls recheck Result type definition 
which is Result<T> it returns T if everything went right and Error if error happened.
I guess &str wont work because its lifetime will end once function returns ?